### ISSUE TITLE
Allow configuring DNS timeout for Http requests

### ISSUE THREAD

**Issue by sm2017** [dart-lang/http#451]
How can I set a DNS resolving timeout?

**Comment by zichangg** [comment]
@sm2017 I don't think you can't set a timeout specific to DNS resolution. `dart:io` and `dart:html` doesn't even provide any API to do that.

**Comment by natebosch** [comment]
Correct. This package does not support this - and it can't unless the browser starts to support it.

You could file a request for supporting this in the SDK, in which case we may be able to support it in the IO client.

**Comment by sm2017** [comment]
@natebosch what about flutter?
